game/views.py

```
@@ -347,7 +354,11 @@ def resign_game(request):
 
     game = ChessGame.from_dict(game_data)
 
-    resigning_player = game.current_turn
+    if game.mode == 'ai':
+        resigning_player = game.player_color
+    else:
+        resigning_player = game.current_turn
```

Ruff (0.15.12)

[warning] Use ternary operator `resigning_player = game.player_color if game.mode == 'ai' else game.current_turn` instead of `if`-`else`-block

Replace `if`-`else`-block with `resigning_player = game.player_color if game.mode == 'ai' else game.current_turn`

(SIM108)